_api-reference/security/configuration/update-configuration.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: default
+title: Update Security Configuration API
+parent: Configuration APIs
+grand_parent: Security APIs
+nav_order: 30
+---
+
+# Update Security Configuration API
+**Introduced 1.0**
+{: .label .label-purple }
+
+The Update Security Configuration API creates or updates the Security plugin's configuration directly through the REST API. This configuration manages core security settings including authentication methods, authorization rules, and access controls.
+
+This operation can easily break your existing security configuration. We strongly recommend using the `securityadmin.sh` script instead, which includes validations and safeguards to prevent misconfiguration.
+{: .warning}
+
+<!-- spec_insert_start
+api: security.update_configuration
+component: endpoints
+-->
+## Endpoints
+```json
+PUT /_plugins/_security/api/securityconfig/config
+```
+<!-- spec_insert_end -->
+
+## Request body fields
+
+The request body is **required**. It is a JSON object with the following fields:
+
+| Property | Required | Data type | Description |
+| :--- | :--- | :--- | :--- |
+| `dynamic` | **Required** | Object | The main configuration object containing all security configuration settings. |
+
+<details markdown="block">
+  <summary>
+    Request body fields: <code>dynamic</code>
+  </summary>
+  {: .text-delta}
+
+`dynamic` is a JSON object with the following fields:
+
+| Property | Data type | Description |
+| :--- | :--- | :--- |
+| `auth_failure_listeners` | Object | Configuration for handling authentication failures, including thresholds and actions. |
+| `authc` | Object | Authentication configuration domains, defining how users are authenticated. |
+| `authz` | Object | Authorization configuration, defining how permissions are evaluated. |
+| `disable_intertransport_auth` | Boolean | When `true`, disables authentication for internal node-to-node communication. |
+| `disable_rest_auth` | Boolean | When `true`, disables authentication for REST API requests (dangerous). |
+| `do_not_fail_on_forbidden` | Boolean | When `true`, returns empty results instead of forbidden error for unauthorized access. |
+| `do_not_fail_on_forbidden_empty` | Boolean | Similar to `do_not_fail_on_forbidden` but with specific behavior for empty results. |
+| `filtered_alias_mode` | String | Controls how document field filtering is applied on aliases. |
+| `hosts_resolver_mode` | String | Determines how hostname resolution is performed for security operations. |
+| `http` | Object | HTTP-specific security configurations. |
+| `kibana` | Object | Legacy settings for Kibana integration (deprecated). |
+| `multi_rolespan_enabled` | Boolean | When `true`, enables spanning permissions across multiple roles. |
+| `on_behalf_of` | Object | Configuration for trusted users to impersonate other users (advanced). |
+| `opensearch-dashboards` | Object | Configuration for OpenSearch Dashboards integration. |
```

Comment:
Does this work? The security plugin still configures dashboards settings with the `kibana` key. It really should be changed, but introduces bwc risk if not done correctly.

Reply:
I haven't tried this myself. Let me look into this.

Reply:
Confirmed this doesn't work. Switching to `kibana`.